python/lsst/pipe/tasks/postprocess.py

```diff
@@ -235,25 +237,62 @@ class WriteObjectTableTask(pipeBase.PipelineTask):
     _DefaultName = "writeObjectTable"
     ConfigClass = WriteObjectTableConfig
 
-    # Names of table datasets to be merged
-    inputDatasets = ("forced_src", "meas", "ref")
+    # Names of table datasets to be merged (is this used anywhere?)
+    # inputDatasets = ("forced_src", "meas", "ref", "psfs_multiprofit", "Sersic_multiprofit")
 
     # Tag of output dataset written by `MergeSourcesTask.write`
     outputDataset = "obj"
 
     def runQuantum(self, butlerQC, inputRefs, outputRefs):
         inputs = butlerQC.get(inputRefs)
 
+        """
+        # Minimally invasive solution
         measDict = {ref.dataId["band"]: {"meas": cat} for ref, cat in
-                    zip(inputRefs.inputCatalogMeas, inputs["inputCatalogMeas"])}
+                    zip(inputRefs.inputCatalogMeas, inputs["inputCatalogMeas"], )}
         forcedSourceDict = {ref.dataId["band"]: {"forced_src": cat} for ref, cat in
                             zip(inputRefs.inputCatalogForcedSrc, inputs["inputCatalogForcedSrc"])}
+        gaussianMixPsfDict = {
+            ref.dataId["band"]: {"gaussianMixPsf": cat}
+            for ref, cat in zip(
+                inputRefs.inputCatalogGaussianMixPsf,
+                inputs["inputCatalogGaussianMixPsf"],
+                strict=True,
+            )
+        }
 
         catalogs = {}
         for band in measDict.keys():
-            catalogs[band] = {"meas": measDict[band]["meas"],
-                              "forced_src": forcedSourceDict[band]["forced_src"],
-                              "ref": inputs["inputCatalogRef"]}
+            catalogs[band] = {
+                "meas": measDict[band]["meas"],
+                "forced_src": forcedSourceDict[band]["forced_src"],
+                "ref": inputs["inputCatalogRef"],
+                "gaussianMixPsf": gaussianMixPsfDict[band],
+                "multibandSersic": inputs["inputCatalogMultibandSersic"],
+            }
+        """
+        datasets_band = defaultdict(dict)
+        for dataset, connection in (
+            ("meas", "inputCatalogMeas"),
+            ("forced_src", "inputCatalogForcedSrc"),
+            ("psfs_multiprofit", "inputCatalogPsfsMultiprofit"),
```

Comment:
Does this mean the plan is to put the multiprofit PSF fits into `deepCoadd_obj` (since they're per-band in the same way as other existing things), but the galaxy fits enter at the transform stage? 

(That'd be fine, but it might merit some commenting, and I was curious whether it was an oversight.)

Reply:
Yes. I tried to merge the Sersic fit table here as well - the same way as for ref - but the existing solution of adding multiband tables to each per-band level in the multi-index is both confusing and inefficient (I'm pretty sure it makes a separate copy per band - at least, the columns can be assigned to independently), so I'd rather just move away from it now.

Reply:
Also afaict nothing relies on `deepCoadd_obj` existing, and I was under the impression that we shouldn't depend on intermediate datasets unless it's necessary, which it really isn't here.